include/onnxruntime/core/common/common.h

```diff
@@ -137,16 +140,25 @@ std::vector<std::string> GetStackTrace();
   ORT_DISALLOW_COPY_AND_ASSIGNMENT(TypeName);           \
   ORT_DISALLOW_MOVE(TypeName)
 
-#define ORT_RETURN_IF_ERROR(expr)          \
-  do {                                     \
-    auto _status = (expr);                 \
-    if ((!_status.IsOK())) return _status; \
+#define ORT_RETURN_IF_ERROR_SESSIONID(expr, session_id)  \
+  do {                                       \
+    auto _status = (expr);                   \
+    if ((!_status.IsOK())) {                 \
+      LogRuntimeError(session_id, _status, __FILE__, __FUNCTION__, __LINE__); \
+      return _status;                        \
+    }                                        \
   } while (0)
 
+#define ORT_RETURN_IF_ERROR_SESSIONID_(expr) ORT_RETURN_IF_ERROR_SESSIONID(expr, session_id_)
+#define ORT_RETURN_IF_ERROR(expr) ORT_RETURN_IF_ERROR_SESSIONID(expr, 0)
+
 #define ORT_THROW_IF_ERROR(expr)               \
```

Comment:
Do we plan to send events for cases when we just do ORT_THROW?

Reply:
good catch.  I first coded ORT_RETURN_IF and ORT_THROW_IF .  it covers most all code paths, but not all.   how about i file a tracking item in a second PR to catch all of the macros?

---
In reply to: [335302117](https://github.com/microsoft/onnxruntime/pull/2109#discussion_r335302117) [](ancestors = 335302117)

Reply:
Sure, that's fine.

Reply:
added as a work item

---
In reply to: [335777787](https://github.com/microsoft/onnxruntime/pull/2109#discussion_r335777787) [](ancestors = 335777787)